src/auto_note.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+#ifndef AUTO_NOTE_H
+#define AUTO_NOTE_H
+
+#include <string>
+#include <unordered_set>
+#include <utility>
+#include <unordered_map>
+#include <vector>
+
+#include "string_id.h"
+#include "map_extras.h"
+
+
+namespace auto_notes
+{
+/**
+ * Class implementing the user interface for the auto note configuration.
+ */
+class auto_note_manager_gui
+{
+    public:
+        auto_note_manager_gui();
+
+        void show();
+
+        bool was_changed() const;
+
+    private:
+        /// The map extra type cache. This is initialized with all known map extra types
+        /// and their auto note status with every call of initialize(). All changes to this
+        /// will be applied to the per-character auto notes settings object after the user
+        /// closes the GUI.
+        std::unordered_map<std::string, std::pair<const map_extra, bool>> mapExtraCache;
+
+        /// All map extra types that will be displayed in the GUI.
+        std::vector<std::string> displayCache;
```

Comment:
```suggestion
        std::vector<string_id<map_extra>> displayCache;
```

And the map above should also have this as key.